net/url.go

```diff
@@ -251,3 +252,21 @@ func IsNetworkOrHostDown(err error, expectTimeouts bool) bool {
 	}
 	return false
 }
+
+// IsConnResetErr - Checks for "connection reset" errors.
+func IsConnResetErr(err error) bool {
+	if strings.Contains(err.Error(), "connection reset by peer") {
+		return true
+	}
+	// incase if error message is wrapped.
+	return errors.Is(err, syscall.ECONNRESET)
+}
+
+// IsConnRefusedErr - Checks for "connection refused" errors.
+func IsConnRefusedErr(err error) bool {
+	if strings.Contains(err.Error(), "connection refused") {
+		return true
+	}
+	// incase if error message is wrapped.
+	return errors.Is(err, syscall.ECONNREFUSED)
```

Comment:
When wouldn't the error type of syscall? You would need to deal with `errors.Unwrap()` if needed

Reply:
Just because `errors.Is` traverses and catches the `syscall.Error`, it is used. `errors.Unwrap()` does only one level check right and it won't go beyond that?

Also, this function is moved from `minio/minio` for common usage.

Reply:
> When wouldn't the error type of syscall

These utils were being used in event store mechanism